worker.go

```diff
@@ -242,7 +262,10 @@ func assignThreadToWorker(thread *phpThread) {
 	if !workersAreReady.Load() {
 		workersReadyWG.Done()
 	}
-	// TODO: we can also store all threads assigned to the worker if needed
+	thread.requestChan = make(chan *http.Request)
+	worker.threadMutex.Lock()
+	defer worker.threadMutex.Unlock()
```

Comment:
`defer` doesn't seem very useful here and may cost one extra allocation. Maybe can we just must the call to `Unlock()` after the call to `append()`?

Reply:
Interesting, I didn't know defer allocates. Changed it 👍